tests/unit/s2n_extensions_server_key_share_select_test.c

```diff
@@ -164,412 +164,412 @@ int main() {
-            EXPECT_NOT_NULL(ecc_pref);
-
-            const struct s2n_kem_preferences *kem_pref = NULL;
-            EXPECT_SUCCESS(s2n_connection_get_kem_preferences(server_conn, &kem_pref));
-            EXPECT_NOT_NULL(kem_pref);
-
-            server_conn->secure.server_ecc_evp_params.negotiated_curve = ecc_pref->ecc_curves[0];
-            server_conn->secure.server_kem_group_params.kem_group = kem_pref->tls13_kem_groups[0];
-
-            EXPECT_FAILURE_WITH_ERRNO(s2n_extensions_server_key_share_select(server_conn),
-                    S2N_ERR_ECDHE_UNSUPPORTED_CURVE);
-
-            EXPECT_FALSE(s2n_is_hello_retry_handshake(server_conn));
-
-            EXPECT_SUCCESS(s2n_connection_free(server_conn));
+        struct s2n_connection *server_conn = NULL;
+        EXPECT_NOT_NULL(server_conn = s2n_connection_new(S2N_SERVER));
+        server_conn->security_policy_override = &test_security_policy;
+        EXPECT_SUCCESS(s2n_connection_set_all_protocol_versions(server_conn, S2N_TLS13));
+
+        const struct s2n_ecc_preferences *ecc_pref = NULL;
+        EXPECT_SUCCESS(s2n_connection_get_ecc_preferences(server_conn, &ecc_pref));
+        EXPECT_NOT_NULL(ecc_pref);
+
+        const struct s2n_kem_preferences *kem_pref = NULL;
+        EXPECT_SUCCESS(s2n_connection_get_kem_preferences(server_conn, &kem_pref));
+        EXPECT_NOT_NULL(kem_pref);
+
+        server_conn->secure.server_ecc_evp_params.negotiated_curve = ecc_pref->ecc_curves[0];
+        server_conn->secure.server_kem_group_params.kem_group = kem_pref->tls13_kem_groups[0];
+
+        EXPECT_FAILURE_WITH_ERRNO(s2n_extensions_server_key_share_select(server_conn),
+                S2N_ERR_ECDHE_UNSUPPORTED_CURVE);
+
+        EXPECT_FALSE(s2n_is_hello_retry_handshake(server_conn));
+
+        EXPECT_SUCCESS(s2n_connection_free(server_conn));
+    }
+
+    /* If client and server have no mutually supported groups but server received a KEM group key share,
+     * a Hello Retry Request flag is not set and the server ignores the keyshare. */
+    {
+        struct s2n_connection *server_conn = NULL;
+        EXPECT_NOT_NULL(server_conn = s2n_connection_new(S2N_SERVER));
+        server_conn->security_policy_override = &test_security_policy;
+        EXPECT_SUCCESS(s2n_connection_set_all_protocol_versions(server_conn, S2N_TLS13));
+
+        const struct s2n_kem_preferences *kem_pref = NULL;
+        EXPECT_SUCCESS(s2n_connection_get_kem_preferences(server_conn, &kem_pref));
+        EXPECT_NOT_NULL(kem_pref);
+
+        /* Server would have not chosen any group when processing the supported_groups extension */
+        EXPECT_NULL(server_conn->secure.server_ecc_evp_params.negotiated_curve);
+        EXPECT_NULL(server_conn->secure.server_kem_group_params.kem_group);
+
+        /* Received an erroneous keyshare for kem group 0 */
+        struct s2n_kem_group_params *client_params0 = &server_conn->secure.client_kem_group_params[0];
```

Comment:
Yeah, the numerical suffix in the variable name indicates which KEM group the parameters correspond to in the list in the KEM preferences. It's not particularly relevant for this specific test since there's only one set of params, but it's consistent with how the rest of the tests are written.